### PR TITLE
feat(payment): PI-1368 SquareV2 - fix types for payment submit payload

### PR DESCRIPTION
**Pull request by PavlenkoM**
## What?
Change payment submitting payload

## Why?
To prevent multiple type guard checks and code duplications.

## Testing / Proof
manually tested and unit test

@bigcommerce/team-checkout @bigcommerce/team-payments

### CODE REVIEW

File: packages/squarev2-integration/src/squarev2-payment-strategy.ts [hunk 1]
```diff
@@ -142,19 +104,54 @@ export default class SquareV2PaymentStrategy implements PaymentStrategy {
...
+            // for each 'verifyBuyer' we need to generate new token
+            const tokenForSavingCard = await this._squareV2PaymentProcessor.tokenize();
+
+            savingCreditCardTokens = {
+                store_card_nonce: tokenForSavingCard,
+                store_card_token: await this._squareV2PaymentProcessor.verifyBuyer(
+                    tokenForSavingCard,
+                    SquareIntent.STORE,
+                ),
+            };
```
Comment (serhii-tkachenko) [inline packages/squarev2-integration/src/squarev2-payment-strategy.ts:138]: It looks good to me, but I wan to suggest another variant:

```suggestion
        let tokenData = {
            nonce: cardTokenizationResult,
            token: await this._squareV2PaymentProcessor.verifyBuyer(
                cardTokenizationResult,
                SquareIntent.CHARGE,
            ),
        };

        if (shouldSaveInstrument) {
            // INFO: additional 'tokenize' is required to verify and save the card
            // for each 'verifyBuyer' we need to generate new token
            const tokenForSavingCard = await this._squareV2PaymentProcessor.tokenize();

            tokenData = {
                ...tokenData,
                store_card_nonce: tokenForSavingCard,
                store_card_token: await this._squareV2PaymentProcessor.verifyBuyer(
                    tokenForSavingCard,
                    SquareIntent.STORE,
                ),
            };
```

File: packages/squarev2-integration/src/squarev2-payment-strategy.ts [hunk 2]
```diff
@@ -142,19 +104,54 @@ export default class SquareV2PaymentStrategy implements PaymentStrategy {
...
         }
 
+        return {
+            credit_card_token: {
+                token: JSON.stringify({
+                    ...creditCardTokens,
+                    ...savingCreditCardTokens,
+                }),
+            },
```
Comment (serhii-tkachenko) [inline packages/squarev2-integration/src/squarev2-payment-strategy.ts:147]: And here you might have: 
```suggestion
            credit_card_token: {
                token: JSON.stringify(tokenData),
            },
```

**Review by serhii-tkachenko — APPROVED** [review]
Nice work! 👍

File: packages/squarev2-integration/src/squarev2-payment-strategy.ts (hunk 1 above)
Comment (PavlenkoM) [reply packages/squarev2-integration/src/squarev2-payment-strategy.ts]: I will apply your variant, thanks!

**Review by vitalii-koshovyi — APPROVED** [review]

**Pull request merged**